libmamba/src/solv-cpp/repo.cpp

```diff
@@ -83,34 +83,74 @@ namespace mamba::solv
 
     namespace
     {
-        struct FilePtr
+        class FilePtr
         {
-            std::FILE* ptr = nullptr;
-            std::string_view name = {};
+        public:
 
-            ~FilePtr() noexcept(false)
+            /**
+             * Open a file with C API.
+             *
+             * @param path must hae filesystem default encoding.
+             */
+            static auto open(std::filesystem::path path, const char* mode) -> FilePtr;
+
+            ~FilePtr() noexcept(false);
+
+            auto raw() noexcept -> std::FILE*;
+            auto raw() const noexcept -> const std::FILE*;
+
+        private:
+
+            FilePtr(std::FILE* ptr, std::string name);
+
+            std::FILE* m_ptr = nullptr;
+            std::string m_name = {};
+        };
+
+        FilePtr::FilePtr(std::FILE* ptr, std::string name)
+            : m_ptr{ ptr }
+            , m_name{ name }
+        {
+        }
+
+        auto FilePtr::open(std::filesystem::path path, const char* mode) -> FilePtr
+        {
+            std::string name = path;
```

Comment:
This doesnt work cross-platform as it's translated back to whatever is the default encoding, resulting in giberish if on Windows with utf-8 non-ascii characters in the path.
To fix it the simplest way is to use `mamba::fs::path` instead of `std::filesystem::path`

Reply:
You can use `mamba::fs::path::std_path()` to get a `std::filesystem::path`